src/main/java/org/gosulang/gradle/tasks/compile/GosuCompile.java

```diff
@@ -190,7 +190,7 @@ private DefaultGosuCompileSpec createSpec() {
     if (_orderClasspath == null) {
       spec.setClasspath(asList(getClasspath()));
     } else {
-      spec.setClasspath(asList(_orderClasspath.call(project, project.getConfigurations().getByName(JavaPlugin.COMPILE_CONFIGURATION_NAME))));
+      spec.setClasspath(asList(_orderClasspath.call(project, project.getConfigurations().getByName(JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME))));
```

Comment:
Consider using `compileClasspath` configuration instead to support "implementation" (and also "api" if java-library plugin is applied to the project), see https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_plugin_and_dependency_management